oauth2-server/src/main/java/com/clouway/oauth2/token/IdTokenBuilder.java

```diff
@@ -0,0 +1,78 @@
+package com.clouway.oauth2.token;
+
+import com.clouway.oauth2.DateTime;
+import com.clouway.oauth2.Identity;
+import com.clouway.oauth2.client.ClientKeyStore;
+import com.clouway.oauth2.jws.Pem;
+import com.clouway.oauth2.jws.Pem.Block;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * @author Vasil Mitov <vasil.mitov@clouway.com>
+ */
+public class IdTokenBuilder implements TokenBuilder {
+  private final ClientKeyStore clientKeyStore;
+
+  public IdTokenBuilder(ClientKeyStore clientKeyStore) {
+    this.clientKeyStore = clientKeyStore;
+  }
+
+  @Override
+  public String issueToken(String host, String clientId, Identity identity, Long ttl, DateTime instant) {
```

Comment:
It's not issuing as it doesn't generate any id or something.  It look likes it creates it.